### PR TITLE
fix: Button Group, MenuButton, Tabs undo Redo fixed.

### LINKED ISSUE #10175: [Bug]: Redo is not happening after deleting a button from button group
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It is observed that when the user deletes any button from the button group property and clicking on cmd+z shows the deleted button in the property pane but not in the widget 

[LOOM DEMO](https://www.loom.com/share/9c76f0baf13e4cb5a5df3eb27ae0b96b)

### Steps To Reproduce

1.Launch appmsith app in release/Production environment 
2.Drag and drop button group widget 
3.Open property pane 
4.Delete any of the button 
5.Press cmd+Z
7.Check for the deleted button in the widget 

### Environment

Production

### Version

Cloud

### LINKED ISSUE #11500: [Bug]: Shuffling of menu items are not reflected until page refresh
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

[LOOM DEMO](https://www.loom.com/share/73a849fe78704aeca616ae87b0e54804)

### Steps To Reproduce

1. Add Menu Widget to canvas
2. Shuffle menu items in different order 
3. Click on menu widget and observe that Items are not re-arranged as expected

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by yaldram**
## Description
- `Button Group`, `MenuButton` and `Tabs Widget` Undo - Redo issues Fixed.
- Also for the `ButtonGroup` and `MenuButton` widgets the re-ordering was not working.
- Also, the property pane navigation for the `MenuItems` and `Tabs` was not working.
- Last, the menuItems and property pane navigation in the Table column.

Fixes #10175 
Fixes #11500

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Tested Manually on the Canvas.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
[LOOM DEMO](https://www.loom.com/share/20c14c73fc8d4b2a9065241b65e04f00)

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/button-group-undo 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.18 **(0.01)** | 37.68 **(0.01)** | 35.66 **(0.01)** | 56.42 **(0.01)**
 :green_circle: | app/client/src/components/propertyControls/ButtonListControl.tsx | 33.33 **(1.82)** | 27.27 **(0)** | 10.53 **(-0.58)** | 32.88 **(1.89)**
 :green_circle: | app/client/src/components/propertyControls/MenuItemsControl.tsx | 31.51 **(1.93)** | 22.22 **(1.53)** | 10.53 **(-0.58)** | 30.99 **(2)**
 :green_circle: | app/client/src/components/propertyControls/TabControl.tsx | 58.67 **(6.17)** | 45 **(6.11)** | 42.11 **(7.11)** | 58.11 **(6.21)**
 :green_circle: | app/client/src/sagas/WidgetAdditionSagas.ts | 70.07 **(0.5)** | 66.67 **(5.45)** | 62.5 **(0)** | 71.07 **(0.58)**
 :green_circle: | app/client/src/widgets/MenuButtonWidget/component/index.tsx | 26.23 **(0.81)** | 11.76 **(0)** | 0 **(0)** | 31.37 **(0.76)**
 :green_circle: | app/client/src/widgets/TableWidget/component/components/menuButtonTableComponent.tsx | 32 **(0.75)** | 14.71 **(0)** | 0 **(0)** | 37.21 **(0.62)**</details>

### CODE REVIEW

**Comment by yaldram** [comment]
Thanks @rahulramesha for debugging this issue with me. Thanks @arunvjn for debugging this issue with me and suggesting the  `this.updateProperties()`  function call.

**Comment by yaldram** [comment]
@rahulramesha you can see we are using `orderBy` multiple times in our code to order the items by index, this is done both under the `Property Pane's Items Draggable List` & `the widgets`. Can we not order the items while putting them into the `DSL (Redux State)` it will reduce the number of `orderBy` items call. It will also prevent any future mistakes if a code change happens and we forget to order the items. Please share your thoughts.

**Comment by yaldram** [comment]
> @rahulramesha you can see we are using `orderBy` multiple times in our code to order the items by index, this is done both under the `Property Pane's Items Draggable List` & `the widgets`. Can we not order the items while putting them into the `DSL (Redux State)` it will reduce the number of `orderBy` items call. It will also prevent any future mistakes if a code change happens and we forget to order the items. Please share your thoughts.

Not possible at this point in time, we might need to sort the items at the undo stage, this needs investigation and also need to take into account other drag and drog properties like Table columns, etc.

File: app/client/src/widgets/ButtonGroupWidget/component/index.tsx [hunk 1]
```diff
@@ -80,9 +80,9 @@ const ButtonGroupWrapper = styled.div<ThemeProp & WrapperStyleProps>`
       : "none"} !important;
 `;
 
-const MenuButtonWrapper = styled.div`
+const MenuButtonWrapper = styled.div<{ renderMode: RenderMode }>`
   flex: 1 1 auto;
-
+  ${({ renderMode }) => renderMode === RenderModes.CANVAS && `height: 100%`};
```
Comment (rahulramesha) [inline app/client/src/widgets/ButtonGroupWidget/component/index.tsx:85]: could you check if this is necessary, If so, why only in Canvas?

**Review by rahulramesha — COMMENTED** [review]
Most of it looks good to me, just have couple of doubts. Also could you please try to add some test cases for the fix.

File: app/client/src/widgets/ButtonGroupWidget/component/index.tsx (hunk 1 above)
Comment (yaldram) [reply app/client/src/widgets/ButtonGroupWidget/component/index.tsx:85]: Yes the styles were breaking in canvas mode, I worked on this feature I will also push this change in the other fix which I am working on.

File: app/client/src/components/propertyControls/BaseControl.tsx [hunk 2]
```diff
@@ -9,10 +9,7 @@ import { PropertyPaneControlConfig } from "constants/PropertyControlConstants";
 import { CodeEditorExpected } from "components/editorComponents/CodeEditor";
 
 // eslint-disable-next-line @typescript-eslint/ban-types
-abstract class BaseControl<P extends ControlProps, S = {}> extends Component<
-  P,
-  S
-> {
+class BaseControl<P extends ControlProps, S = {}> extends Component<P, S> {
```
[the remark this answers is not in the document]
Comment (rahulramesha) [reply app/client/src/components/propertyControls/BaseControl.tsx:12]: Got it, was just curious, cuz it looked unrelated

**Review by rahulramesha — APPROVED** [review]
LGTM, Try adding cypress test wherever possible.

**Comment by vivekverma2312** [comment]
@yaldram Tested and working fine.

**Pull request merged**